pkg/client/testingclient/reactive.go

```diff
@@ -0,0 +1,243 @@
+package testingclient
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
```

Comment:
We most certainly do not want to add ginkgo or gomega in any non-test files, downstream projects should be able to decide what testing framework they do or do not want to use

Reply:
Good point. The uses here are to make some precondition assertions. Do you think it's reasonable to raise a `panic()`, or is it better to bubble up an `error`? My hesitation for returning an error is that it could be hard to diagnose the issue if the error doesn't make it back to the test. My hesitation with panicking is that it will stop all further tests, and the output isn't very friendly. Nevertheless, I'm leaning towards panicking.